core/sile.lua

```diff
@@ -132,6 +132,7 @@ Options:
   -- http://lua-users.org/wiki/VarargTheSecondClassCitizen
   local identity = function (...) return unpack({...}, 1, select('#', ...)) end
   SILE.errorHandler = opts.traceback and debug.traceback or identity
+  SILE.detailedErrors = not not opts.traceback
```

Comment:
`not not`? Are you trying to force a boolean? Maybe use SU.boolean() instead. Or just use it as is if you assume Lua knows how to cast it already.

Reply:
Yes, this is just to force a boolean. I didn't know about `SU.boolean`, but I don't really need that logic here (I think). And I didn't want to keep it as it is, because then it creates a weird API, which does not communicate intent. (And that mindset leads to problems in the long run, e.g. with GC.)

Cast to boolean would be better, but Lua doesn't have that (or at least I couldn't find it).

Reply:
```suggestion
  SILE.traceback = opts.traceback
```

I would avoid calling it one thing in the user facing opts and a different thing internally, if the user facing name is a good pick then using that as the internal representation should be a good pick too. As for forcing the variable type, I don't think you need to at all. Everywhere it's used it is tested as a boolean and if the casting works here it will work everywhere else.

Reply:
Ok. But I still think that keeping this as a boolean leads to a more predictable and cleaner code.